### ISSUE TITLE
Can't reproduce the result in paper

### ISSUE THREAD

**Issue by Xiaoqiong-Liu** [wimmerth/anyup#15]
**My training environment** 
**Upsampler Training:**
Code: anyup repo (no local code edits), run via python train.py.
Config: default config/base.yaml; 
Backbone: anyup.backbone.PretrainedViTWrapper with vit_small_patch14_dinov2.lvd142m, norm=True.
Model params: qk_dim=128, kernel_size=1, LFU kernel=5, window_ratio=0.1, 4 heads.
Optimizer: AdamW (lr=2e-4, betas 0.9/0.999, weight_decay=1e-5).
Data: ImageNet from {project}/data/imagenet, train/val splits cached under data/cache.
Dataloaders: train batch size 4, val batch size 16, 16 workers, pin_memory=True, seed_worker.
Global settings: epochs=4, max_steps=100_000, img_size=224, crop_size=448, num_local_crops=4, downsampling_regularization=0.1, augmentation_strength=1.0, augmentation_regularization=0.1, sanity=False.
Training log: output/2025-11-17/16-03-46/train.log (begins 2025‑11‑17 16:03:46, shows dataset size 154,884 and periodic loss entries). Checkpoints written to output/2025-11-17/16-03-46/version_0/model_{0,20000,...}.pth.

**Probe evaluation setup**
Command: python evaluation/train_probes.py eval.task=seg dataset_evaluation=voc backbone.name=vit_small_patch14_dinov2.lvd142m eval.model_ckpt={project}/2025-11-17/16-03-46/version_0/model_100000steps.pth optimizer.lr=5e-4.
Scheduler change: CosineAnnealingLR now steps once per epoch (was once per batch).
Batch size tested: 4 (per author recommendation) and default; both give ~±1% difference.
Result (after scheduler tweak): accuracy 0.9268, IoU 0.7311 on VOC (evaluation/unsupervised/voc/vit_small_patch14_dinov2.lvd142m/train_anyup_seg.log).

**Observed behavior:** LR scheduling or batch size tweaks only slightly affect metrics; suspect the pretrained upsampler itself underperforms.

**Questions**:
Whether the released config (4 epochs, LR 2e‑4, batch size 4) matches the paper training recipe. 
Any other ideas?

[config.yaml](https://github.com/user-attachments/files/23637765/config.yaml)

**Comment by wimmerth** [comment]
Hi, thanks for your interest in AnyUp.
As per my last comment in issue #14 , I am asking you to stick to the evaluation setting described in JAFAR, ie. use 20 epochs for linear probe training on VOC and ADE20k datasets.
You need to set this yourself when running the probe training (ie, set num_epochs=20 in eval.yaml).

**Comment by wimmerth** [comment]
Hm, that sounds strange. How about reproduction of the results on other datasets? Another way to test this would be to simply train the probes for JAFAR or LoftUp in your setup and check if these results match the reported ones.

**Comment by wimmerth** [comment]
To add up on this: Were you able to reproduce the evaluation results with the pre-trained network that you can load via torch.hub?

**Comment by Xiaoqiong-Liu** [comment]
I have trained the probes for reproducedJAFAR upsampler, it's around mIoU 0.839 for VOC dataset.  Haven't try the torch.hub yet, will try it. Another idea is to add feature-agnositic layer(can be seen as dictionary learning?) to JAFAR, and see how the result goes

**Comment by Xiaoqiong-Liu** [comment]
Another thing, when I ran jafar train_probes.py, I will get error ' Key 'feature_dim' is not in struct', so I fall back to backbone.embed_dim, I am not sure if this fix is correct or wrong

**Comment by Xiaoqiong-Liu** [comment]
> To add up on this: Were you able to reproduce the evaluation results with the pre-trained network that you can load via torch.hub?

I got Results: {'accuracy': 0.9235197901725769, 'iou': 0.722396969795227} for torch.hub
So my prober is not working well. Could you please share your train_probes.py? Thanks.

**Comment by wimmerth** [comment]
Hi, I also ran the probing experiment (with a newly cloned JAFAR repo) with the torch hub model and I have good and bad news for you. The good news is that **AnyUp has the same results as reported in the paper**. The bad news is that there is something wrong with your probe training then. I attached all files that I changed / added. Please retry with these files.
Likely, the issue is that you're using `backbone.embed_dim` instead of using the actual feature dimensionality of the feature extractor, i.e., `feature_dim=384` for DINOv2 ViT-S.

[base.yaml](https://github.com/user-attachments/files/23647022/base.yaml)
[anyup.yaml](https://github.com/user-attachments/files/23647023/anyup.yaml)
[eval.yaml](https://github.com/user-attachments/files/23647024/eval.yaml)
[anyup.py](https://github.com/user-attachments/files/23647021/anyup.py)
[train_probes.py](https://github.com/user-attachments/files/23647025/train_probes.py)

**Comment by Xiaoqiong-Liu** [comment]
solved by change a line for setup model  [train_probel.py](https://github.com/PaulCouairon/JAFAR/blob/main/evaluation/train_probes.py) :
```python
# model.load_state_dict(checkpoint["jafar"], strict=False)
model.load_state_dict(checkpoint, strict=False)
```
Thanks for providing the details and help
